scripts/updateModHistory.php

```diff
@@ -0,0 +1,219 @@
+
+$fp = fopen($modHistoryFile, "w");
+fwrite($fp, "<?php\n\n/* This is a generated file */\n\nreturn [\n");
+foreach ($mergedModHistory as $fileName => $fileProps) {
+    if ($fileName === "last commit hash") {
+        fwrite($fp, "    \"last commit hash\" => \"" . implode("", $fileProps) . "\",\n");
+        continue;
+    }
+    $newModHistoryString = '    "' . $fileName . "\" => [\n";
+    $newModHistoryString .= "        \"modified\" => \"" . ($fileProps["modified"] ?? "") . "\",\n";
+    $newModHistoryString .= "        \"contributors\" => [\n";
+    if (isset($fileProps["contributors"])) {
+        if (!is_array($fileProps["contributors"])) {
+            exit("Non-array contributors list\n");
+        }
+        foreach ($fileProps["contributors"] as $contributor) {
+            $newModHistoryString .= "            \"" . $contributor . "\",\n";
+        }
+    }
+    $newModHistoryString .= "        ],\n";
+    $newModHistoryString .= "    ],\n";
+    fwrite($fp, $newModHistoryString);
+}
+fwrite($fp, "];\n");
+fclose($fp);
+
+echo "done at " . date('H:i:s') . "\n";
+
+function timeStamp(): string {
+    return "[" . date('H:i:s') . "]";
+}
+
+function processGitDiffLine($line, &$modifiedFiles): void {
+    static $currentType = "";
+    static $currentFile = "";
+    global $fileCounter;
+
+    switch ($line) {
+        case "filename:":
+            $currentType = "filename";
+            $fileCounter++;
+            return;
+        case "modified:":
+            $currentType = "modDateTime";
+            return;
+        case "contributors:":
+            $currentType = "contributors";
+            return;
+        case "last commit hash:":
+            $currentType = "commitHash";
+            return;
+    }
+    if ($currentType === "") {
+        return;
+    }
+
+    switch ($currentType) {
+        case "filename":
+            $currentFile = $line;
+            break;
+        case "modDateTime":
+            if ($currentFile === "") {
+                return;
+            }
+            $modifiedFiles[$currentFile]["modified"] = $line;
+            break;
+        case "contributors":
+            if ($currentFile === "") {
+                return;
+            }
+            $modifiedFiles[$currentFile]["contributors"][] = htmlspecialchars($line, ENT_QUOTES | ENT_SUBSTITUTE | ENT_HTML401);
+            break;
+        case "commitHash":
+            $modifiedFiles["last commit hash"][] = $line;
+            break;
+    }
+}
+
+function verifyCommandLineOptions($commandLineOptions): void {
+    echo timeStamp() . " - Parsing command line arguments... ";
+    if ($commandLineOptions === false
+        || !isset($commandLineOptions["docs-path"])) {
+        echo "\"--docs-path\" is a required argument\n";
+        exit(1);
+    }
+    echo "documentation path supplied\n";
+    if (isset($commandLineOptions["history-path"])) {
+        echo "                                               mod history file path supplied\n";
```

Comment:
Maybe have a variable for the space indent?

Reply:
I used `strlen()` in my last commit. Will that work too or would a variable be better in this case?